### PR TITLE
ceph-osd: ceph-volume container support

### PR DESCRIPTION
**Pull request by leseb**
Signed-off-by: Sébastien Han <seb@redhat.com>

Closes: https://github.com/ceph/ceph-ansible/issues/2717

todo:

- [x] add code to support ceph-volume in container on the ceph_volume module

- [x] add code to support ceph-volume in container in the playbook

- [x] implement batch command in container

- [x] add doc

- [x] add warning flag explaining as of 3.2 ceph-disk won't be supported anymore (code will remain though), the ceph-disk code will go away on 3.3

- [x] add new functionnal tests for ceph-volume container

Remaining TODO in another PR:

- [ ] add support in shrink-osd.yml container for ceph-volume

- [x] add support in purge-docker-cluster.yml https://github.com/ceph/ceph-ansible/pull/2866

### CODE REVIEW

File: library/ceph_volume.py [hunk 1]
```diff
@@ -153,16 +177,12 @@ def create_osd(module):
     wal_vg = module.params.get('wal_vg', None)
     crush_device_class = module.params.get('crush_device_class', None)
     dmcrypt = module.params['dmcrypt']
+    containerized = module.params.get('containerized')
+    subcommand = "create"
```
Comment (andrewschoen) [inline library/ceph_volume.py]: Shouldn't this be ``subcommand = "prepare"``?

File: library/ceph_volume.py [hunk 2]
```diff
@@ -334,11 +359,17 @@ def run_module():
     action = module.params['action']
 
     if action == "create":
-        create_osd(module)
+        prepare_osd(module)
+        activate_osd(module)
+    elif action == "prepare":
+        activate_osd(module)
```
Comment (andrewschoen) [inline library/ceph_volume.py]: This should be``prepare_osd(module)``

File: library/ceph_volume.py (hunk 1 above)
Comment (leseb) [reply library/ceph_volume.py]: @andrewschoen thanks for the review even though this wasn't ready for review :), will address your comments soon.

**Comment by leseb** [comment]
jenkins test luminous-ansible2.4-lvm_osds

**Comment by leseb** [comment]
jenkins test luminous-ansible2.4-lvm_osds

**Comment by leseb** [comment]
jenkins test luminous-ansible2.4-lvm_osds

**Comment by leseb** [comment]
jenkins test luminous-ansible2.4-lvm_osds

**Comment by leseb** [comment]
jenkins test luminous-ansible2.4-lvm_osds

**Comment by leseb** [comment]
jenkins test luminous-ansible2.4-lvm_osds

**Comment by leseb** [comment]
jenkins test luminous-ansible2.4-lvm_osds

**Comment by leseb** [comment]
jenkins test luminous-ansible2.4-lvm_osds

**Comment by leseb** [comment]
jenkins test luminous-ansible2.4-lvm_osds

**Comment by leseb** [comment]
jenkins test luminous-ansible2.4-lvm_osds

**Comment by leseb** [comment]
jenkins test luminous-ansible2.4-lvm_osds

**Comment by leseb** [comment]
jenkins test luminous-ansible2.4-lvm_osds

**Comment by leseb** [comment]
jenkins test luminous-ansible2.4-lvm_osds

**Comment by leseb** [comment]
jenkins test luminous-ansible2.4-lvm_osds

**Comment by leseb** [comment]
jenkins test luminous-ansible2.4-lvm_osds

**Comment by leseb** [comment]
jenkins test luminous-ansible2.4-lvm_osds

**Comment by leseb** [comment]
This is currently blocked by http://tracker.ceph.com/issues/24952

**Comment by leseb** [comment]
This is currently blocked by http://tracker.ceph.com/issues/24952
@alfredodeza any pointer would be appreciated, thanks.

**Comment by leseb** [comment]
@alfredodeza this is an old comment you're referring too, I just resurrected this so I don't know yet where we are.

**Comment by leseb** [comment]
jenkins test luminous-ansible2.4-lvm_osds

**Comment by leseb** [comment]
jenkins test luminous-lvm_osds

**Comment by leseb** [comment]
jenkins test luminous-lvm_osds

**Comment by leseb** [comment]
jenkins test luminous-lvm_osds

**Comment by leseb** [comment]
jenkins test luminous-lvm_osds

**Comment by leseb** [comment]
jenkins test luminous-lvm_osds

**Comment by leseb** [comment]
jenkins test ceph-volume all

**Comment by leseb** [comment]
jenkins test ceph-volume

**Comment by leseb** [comment]
jenkins test ceph-volume lvm all

**Comment by leseb** [comment]
jenkins test luminous-lvm_batch

**Comment by leseb** [comment]
jenkins test luminous-lvm_osds

**Comment by leseb** [comment]
jenkins test pipeline

**Comment by leseb** [comment]
jenkins test pipeline

**Comment by leseb** [comment]
@andrewschoen this is ready for final review. I need your input on how to test the batch scenario. Thanks.

**Comment by leseb** [comment]
jenkins test pipeline

**Comment by leseb** [comment]
jenkins test luminous-lvm_osds_container

**Comment by leseb** [comment]
jenkins test luminous-lvm_osds_container

**Comment by leseb** [comment]
jenkins test luminous-lvm_osds_container

**Comment by leseb** [comment]
jenkins test luminous-lvm_osds_container

**Comment by leseb** [comment]
jenkins test luminous-lvm_osds_container

**Comment by guits** [comment]
jenkins test luminous-lvm_osds_container

**Comment by guits** [comment]
jenkins test luminous-lvm_osds_container

**Comment by leseb** [comment]
jenkins test luminous-lvm_osds_container

**Comment by leseb** [comment]
jenkins test luminous-bluestore_lvm_osds_container

**Comment by leseb** [comment]
jenkins test luminous-lvm_osds_container

**Comment by leseb** [comment]
jenkins test luminous-bluestore_lvm_osds_container

**Comment by leseb** [comment]
jenkins test luminous-bluestore_lvm_osds_container

**Comment by leseb** [comment]
jenkins test luminous-lvm_osds_container

**Comment by leseb** [comment]
jenkins test luminous-lvm_osds_container

**Comment by leseb** [comment]
jenkins test luminous-bluestore_lvm_osds_container

**Comment by leseb** [comment]
jenkins test luminous-bluestore_lvm_osds

**Comment by leseb** [comment]
jenkins test luminous-lvm_osds

**Comment by leseb** [comment]
jenkins test luminous-lvm_osds_container

**Comment by leseb** [comment]
jenkins test luminous-bluestore_lvm_osds_container

**Comment by leseb** [comment]
jenkins test luminous-lvm_batch

**Comment by leseb** [comment]
jenkins test luminous-lvm_batch_container

**Comment by leseb** [comment]
jenkins test luminous-lvm_osds_container

**Comment by leseb** [comment]
jenkins test luminous-bluestore_lvm_osds_container

**Comment by leseb** [comment]
jenkins test luminous-lvm_batch

**Comment by leseb** [comment]
jenkins test pipeline

**Comment by leseb** [comment]
Tests for docker_cluster and centos7 will be fixed in https://github.com/ceph/ceph-ansible/pull/3187

**Comment by guits** [comment]
failures are expected, merging anyway

**Pull request merged**